zenodo/modules/records/admin.py

```diff
@@ -0,0 +1,124 @@
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Zenodo; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Admin view for updating datacite metadata."""
+
+from datetime import datetime
+import uuid
+
+from flask import current_app, flash, redirect, request, url_for
+from flask_admin import BaseView, expose
+from flask_babelex import lazy_gettext as _
+from flask_wtf import FlaskForm
+from invenio_cache import current_cache
+from wtforms.fields.html5 import DateField
+from wtforms.validators import DataRequired
+
+from zenodo.config import DATACITE_UPDATING_RATE_PER_HOUR
+from zenodo.modules.records.utils import find_registered_doi_pids
+
+
+class UpdateDataciteForm(FlaskForm):
+    """Form for updating datacite metadata."""
+
+    field_sets = ['from_date', 'until_date']
+
+    from_date = DateField(
+        _('From'),
+        description=_('Required.'),
+        validators=[DataRequired()],
+    )
+
+    until_date = DateField(
+        _('Until'),
+        description=_('Required.'),
+        validators=[DataRequired()],
+    )
+
+
+class UpdateDataciteView(BaseView):
+    """View for updating datacite metadata."""
+
+    @expose('/', methods=('GET', 'POST'))
+    def data_cite(self):
+        """."""
+        form = UpdateDataciteForm()
+        details_form = FlaskForm()
+
+        is_task_running = False
+        time = 0
+        task_details = current_cache.get('update_datacite:task_details')
+
+        if task_details:
+            is_task_running = True
+            if details_form.validate_on_submit():
+                current_cache.set('update_datacite:task_details', None)
+                return redirect(url_for('updatedataciteview.data_cite'))
+        else:
+            if form.validate_on_submit():
+                from_date = request.form['from_date']
+                until_date = request.form['until_date']
+
+                action = request.form['action']
+                if action == 'SubmitDates':
+                    if from_date > until_date:
+                        flash("Error: the 'From' date should precede the 'Until' date.")
+                    else:
+                        pids_count = find_registered_doi_pids(from_date,
+                                                                    until_date,
+                                                                    current_app.config['ZENODO_LOCAL_DOI_PREFIXES']).count()
+                        task_details = dict(
+                            total_pids=pids_count
+                        )
+                        time = pids_count/DATACITE_UPDATING_RATE_PER_HOUR
```

Comment:
It's better to use `current_app.config['DATACITE_UPDATING_RATE_PER_HOUR']`, since this might be overriden by some config in e.g. `invenio.cfg`